runtime/protocol/http/common/src/aws/smithy/kotlin/runtime/http/QueryParameters.kt

```diff
@@ -49,6 +49,8 @@ public fun Map<String, String>.toQueryParameters(): QueryParameters {
 
 private class QueryParametersImpl(values: Map<String, List<String>> = emptyMap()) : QueryParameters, StringValuesMapImpl(true, values) {
     override fun toString(): String = "QueryParameters ${entries()}"
+
+    override fun equals(other: Any?): Boolean = other is QueryParametersImpl && entries() == other.entries()
```

Comment:
**Question**: Why require `other` to be `QueryParametersImpl` vs `QueryParameters`? This means that the following code will fail but believe we'd want it to succeed:

```kotlin
val expected = QueryParameters.Empty
val actual = QueryParameters { }
assertEquals(expected, actual)
```